actionmailer/lib/action_mailer/parameterized.rb

```diff
@@ -131,7 +131,8 @@ def initialize(mailer_class, action, params, *args)
         def processed_mailer
           @processed_mailer ||= @mailer_class.new.tap do |mailer|
             mailer.params = @params
-            mailer.process @action, *@args
+            args = mailer.method(@action).arity.zero? ? [] : @args
```

Comment:
Why we need to check the arity now?

Reply:
I think this works by coincidence and lack of test coverage for parameterized mailers with additional args. Now that I have a better idea of how parameterized mailers work, I'll refactor this to be less hacky.

Reply:
Done. This version also doesn't break the contract of `assert_enqueued_email_with` when using a hash: `If +args+ is provided as a Hash, a parameterized email is matched`.